data-hibernate-jpa/src/main/java/io/micronaut/data/hibernate/operations/AbstractHibernateOperations.java

```diff
@@ -470,7 +470,7 @@ public Object convert(Class<?> converterClass, Object value, Argument<?> argumen
             @Override
             public void bindOne(QueryParameterBinding binding, Object value) {
                 String parameterName = Objects.requireNonNull(binding.getName(), "Parameter name cannot be null!");
-                if (storedQuery.isNative()) {
+                if (binding.getParameterIndex() != -1) {
```

Comment:
This is an attempt to attack a more abstract problem than maybe this issue poses, I thought this code path could be good outside for more than the native query context (as this issue shows)

Reply:
This seems to fix the issue, which I believe is caused in https://github.com/micronaut-projects/micronaut-data/blob/4.8.x/data-runtime/src/main/java/io/micronaut/data/runtime/operations/internal/query/DefaultBindableParametersStoredQuery.java#L202-L207
```
List<Object> values = binding.isExpandable() ? expandValue(value, binding.getDataType()) : null;
        if (values != null && values.isEmpty()) {
            // Empty collections / array should always set at least one value
            value = null;
            values = null;
        }
```
as Micronaut binder didn't expect empty list for IN clause.
@dstepanov not sure if this change could cause some other issues.

Reply:
The emptyList for IN with non-entity types works (at least [basic types](https://jakarta.ee/specifications/persistence/3.2/jakarta-persistence-spec-3.2#a486)), maybe by accident, maybe intentionally doesn't want null, I'm unsure. Hibernate [EntityJavaType::isInstance](https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/type/descriptor/java/spi/EntityJavaType.java#L56) doesn't protect against null, but it's parent [JavaType](https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/type/descriptor/java/JavaType.java#L106) doesn't have the same issue (which this used for basic types)